geoapi/tasks/external_data.py

```diff
@@ -57,47 +66,71 @@ def get_file(client, system_id, path, required):
+        logger.info(f'Determining which shapefile-related files need to be downloaded for file {current_file.filename}')
         for extension, required in SHAPEFILE_FILE_ADDITIONAL_FILES.items():
-            additional_file_path = path.with_suffix(extension)
+            additional_file_path = current_file_path.with_suffix(extension)
             if available_files and str(additional_file_path) not in available_files:
                 if required:
-                    logger.error("Could not import required shapefile-related file: "
-                                 "agave: {} :: {}".format(systemId, additional_file_path))
-                    raise Exception("Required file ({}) missing".format(additional_file_path))
+                    logger.error(f"Could not import required shapefile-related file: agave: {system_id}/{additional_file_path}")
+                    raise Exception(f'Required file ({system_id}/{additional_file_path}) missing')
                 else:
                     continue
-            paths_to_get.append(additional_file_path)
-
-        additional_files = []
-        with concurrent.futures.ThreadPoolExecutor(max_workers=5) as executor:
-            getting_files_futures = [executor.submit(get_file, client, systemId, additional_file_path, required)
-                                     for additional_file_path in paths_to_get]
-            for future in concurrent.futures.as_completed(getting_files_futures):
-                _, additional_file_path, required, result_file, error = future.result()
-                if not result_file and required:
-                    logger.error("Could not import a required shapefile-related file: "
-                                 "agave: {} :: {}   ---- error: {}".format(systemId, additional_file_path, error))
-                if not result_file:
-                    logger.debug("Unable to get non-required shapefile-related file: "
-                                 "agave: {} :: {}".format(systemId, additional_file_path))
-                    continue
-                result_file.filename = Path(additional_file_path).name
-                additional_files.append(result_file)
+            additional_files_to_get.append(AdditionalFile(path=additional_file_path, required=required))
+    elif file_suffix == "rq":
+        logger.info(f'Parsing rq file {current_file.filename} to see what assets need to be downloaded ')
+        data = json.load(current_file)
+        for section in data["sections"]:
+            for question in section["questions"]:
+                for asset in question.get("assets", []):
+                    # determine full path for this asset and add to list
+                    additional_file_path = current_file_path.with_name(asset["filename"])
+                    additional_files_to_get.append(AdditionalFile(path=additional_file_path, required=True))
+        logger.info(f'{len(additional_files_to_get)} assets were found for rq file {current_file.filename}')
```

Comment:
There is an inconsistency with using single and double quotes through this file. Could we normalize that as much as possible to one of the two?
Compare lines 88, 93, 94, 99, 107, 122, etc.

Reply:
I don't know that TACC WMA has a preference but I tend to prefer this usage:

- raw string literals for regular expressions
- single quotes for small symbol-like strings
- double quotes around strings that are used for interpolation or that are natural language messages (break the rules as needed if the strings contain quotes to escape)
- triple double quotes for docstrings

Reply:
I took a stab at this in  65eeef84e6a409bbad1d1eda9a241e33f79b77f7.  let me know if that matches up with your comment.